docs/proposals/31-inplace-node-update.md

```diff
@@ -0,0 +1,388 @@
+3. `machine-controller-manager` will select the nodes based on the configured `maxUnavailable` and label them with `selected-for-update`. This label can be used to identify nodes that are currently undergoing an update.
+4. `machine-controller-manager` will cordon and drain the node and label it with `ready-for-update` label.
+5. `gardener-node-agent` will detect the `ready-for-update` label and perform the update on the node. And it will wait for the update to complete within the specified timeout.
+6. Once the Kubernetes/Machine image version is updated, the node will be labeled by the `gardener-node-agent` with `update-successful` label.
+7. `machine-controller-manager` will `uncordon` the node, and remove all the added labels. The node is then ready to schedule the workloads.
+
+![InplaceUpdate](assets/inplace-update.png "InPlace Update Strategy")
+
+#### InplaceUpdateOnLabel strategy
+
+1. On shoot spec change, nodes will be labeled with `candidate-for-update`.
+2. `machine-controller-manager` will add the necessary annotation to the nodes to prevent it from being scaled down by `cluster-autoscaler` during the update process.
+3. `machine-controller-manager` will wait for the `selected-for-update` label on the node. The orchestration of the update is solely the responsibility of the user. They are free to select the nodes to be updated at will.
+4. Once the `selected-for-update` label is added by the user on the node, steps 4-7 of the `Inplace` strategy will performed.
+
+![InplaceUpdateOnLabel](assets/inplace-update-onlabel.png "InPlace Update OnLabel Strategy")
+
+### Gardener
+
+#### `CloudProfile` API
+
+A new field `minSupportedVersionInPlaceUpdate` is introduced under `.spec.machineImages[]` under the CloudProfile spec. This field can be used to specify the minimum supported version from which the current machine image version can be in-place updated to.
+
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: CloudProfile
+metadata:
+  name: cloudprofile1
+spec:
+  type: <some-provider-name>
+  providerConfig: <some-provider-specific-cloudprofile-config>
+  kubernetes:
+    versions:
+      - version: 1.31.1
+  machineImages:
+    - name: gardenlinux
+      versions:
+        - version: 1443.8.0
+          expirationDate: "2025-02-28T23:59:59Z"
+          minSupportedVersionInPlaceUpdate: 1312.3.0
+        - version: 1443.7.0
+  machineTypes:
+    - name: m5.large
+      cpu: "2"
+      gpu: "0"
+      memory: 8Gi
+      usable: true
+  regions:
+    - name: <region-name>
+```
+
+#### `Shoot` API
+
+A new field `UpdateStrategy` is introduced under `spec.provider.workers[]` in the Shoot spec. This field will be passed on to the worker extension. Once set, changing the `UpdateStrategy` is prohibited through validation. Additionally, when the `InPlaceUpdate/InplaceUpdateOnLabel` update strategy is configured, skipping the worker pool's Kubernetes minor version is not allowed.
```

Comment:
Is there a specific reason why minor version skipping will not be supported with in-place updates?